worker/pkg/workflows/datasync/activities/activities.go

```diff
@@ -599,16 +601,34 @@ func (b *benthosBuilder) buildProcessorConfig(ctx context.Context, cols []*mgmtv
 		}
 	}
 
-	pc := &neosync_benthos.ProcessorConfig{
-		Mutation: strings.Join(mutations, "\n"),
-		Javascript: neosync_benthos.JavascriptConfig{
-			Code: strings.Join(javascript, "\n"),
-		},
+	mutationStr := strings.Join(mutations, "\n")
+
+	pc := &neosync_benthos.ProcessorConfig{}
+	if len(mutationStr) > 0 {
+		pc.Mutation = &mutationStr
+	}
+	if len(javascript) > 0 && !sliceContainsEmptyString(javascript) {
```

Comment:
if youre worried about empty strings, then that should be handled above at the construction step on line 587.